purldb-toolkit/README.rst

```diff
@@ -20,84 +25,541 @@ Use this command to get basic help::
       --file FILENAME    Read a list of PURLs from a FILE, one per line.
       --help             Show this message and exit.
 
+Examples
+########
+
+**Submit multiple PURLs using the command line:**
+
+.. code-block:: none
+
+    purlcli versions --purl pkg:npm/canonical-path --purl pkg:nginx/nginx --output <path/to/output.json>
+
+*Sample output:*
+
+.. code-block:: console
+
+    {
+        "headers": [
+            {
+                "tool_name": "purlcli",
+                "tool_version": "0.2.0",
+                "options": {
+                    "command": "versions",
+                    "--purl": [
+                        "pkg:npm/canonical-path",
+                        "pkg:nginx/nginx"
+                    ],
+                    "--file": null,
+                    "--output": "<path/to/output.json>"
+                },
+                "errors": [],
+                "warnings": [
+                    "'pkg:nginx/nginx' not supported with `versions` command"
+                ]
+            }
+        ],
+        "packages": [
+            {
+                "purl": "pkg:npm/canonical-path@0.0.1",
+                "version": "0.0.1",
+                "release_date": "2013-12-19"
+            },
+            {
+                "purl": "pkg:npm/canonical-path@0.0.2",
+                "version": "0.0.2",
+                "release_date": "2013-12-19"
+            },
+            {
+                "purl": "pkg:npm/canonical-path@1.0.0",
+                "version": "1.0.0",
+                "release_date": "2018-10-24"
+            }
+        ]
+    }
+
 
-- Collect package metadata for a PURL::
+Notes
+#######
+
+``versions`` calls ``versions()`` from `fetchcode/package_versions.py <https://github.com/nexB/fetchcode/blob/master/src/fetchcode/package_versions.py>`__.
+
+Version information is not needed in submitted PURLs and, if included, will be removed before processing.
+
+
+----
+
+
+'metadata' -- collect package metadata for a PURL
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: console
 
     $ purlcli metadata --help
     Usage: purlcli metadata [OPTIONS]
-    
+
       Given one or more PURLs, for each PURL, return a mapping of metadata fetched
       from the fetchcode package.py info() function.
-    
+
     Options:
       --purl TEXT        PackageURL or PURL.
       --output FILENAME  Write meta output as JSON to FILE.  [required]
       --file FILENAME    Read a list of PURLs from a FILE, one per line.
-      --unique           Return data only for unique PURLs.
       --help             Show this message and exit.
 
-
-- Collect package URLs for a PURL::
+Examples
+########
+
+**Submit multiple PURLs using the command line:**
+
+.. code-block:: none
+
+    purlcli metadata --purl pkg:openssl/openssl@3.0.6 --purl pkg:nginx/nginx@0.8.9 --purl pkg:gnu/glibc@2.38 --output <path/to/output.json>
+
+*Sample output:*
+
+.. code-block:: console
```

Comment:
This should be consistent, JSON and console should be used consistently as applicable, for JSON snippets and commands respectively across this page.

Reply:
Good point and done.